From 303d93ea1e4ac6a645e283116350b27e928484c4 Mon Sep 17 00:00:00 2001
From: Sachin Prabhu <sprabhu@redhat.com>
Date: Mon, 28 Jul 2014 12:16:20 -0400
Subject: [fs] cifs: don't spam the logs on unexpected lookup errors

Message-id: <1406549822-10376-28-git-send-email-sprabhu@redhat.com>
Patchwork-id: 86764
O-Subject: [RHEL7.1 PATCH 27/69] [bz1082049] cifs: don't spam the logs on unexpected lookup errors
Bugzilla: 1082049
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@redhat.com>

Andrey reported that he was seeing cifs.ko spam the logs with messages
like this:

    CIFS VFS: Unexpected lookup error -26

He was listing the root directory of a server and hitting an error when
trying to QUERY_PATH_INFO against hiberfil.sys and pagefile.sys. The
right fix would be to switch the lookup code over to using FIND_FIRST,
but until then we really don't need to report this at a level of
KERN_ERR. Convert this message over to FYI level.

Reported-by: "Andrey Shernyukov" <andreysh@nioch.nsc.ru>
Signed-off-by: Jeff Layton <jlayton@redhat.com>
Signed-off-by: Steve French <smfrench@gmail.com>
(cherry picked from commit a8582159edb47a60dd1e5df5a21890f225270f62)

Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/cifs/dir.c b/fs/cifs/dir.c
index f039c23..a514e0a 100644
--- a/fs/cifs/dir.c
+++ b/fs/cifs/dir.c
@@ -757,7 +757,7 @@ cifs_lookup(struct inode *parent_dir_inode, struct dentry *direntry,
  /* if it was once a directory (but how can we tell?) we could do
   shrink_dcache_parent(direntry); */
  } else if (rc != -EACCES) {
-  cifs_dbg(VFS, "Unexpected lookup error %d\n", rc);
+  cifs_dbg(FYI, "Unexpected lookup error %d\n", rc);
   /* We special case check for Access Denied - since that
   is a common return code */
  }
-- 
1.7.1